src/NUnitFramework/framework/Constraints/MsgUtils.cs

```diff
@@ -395,64 +395,83 @@ public static string GetTypeRepresentation(object obj)
         [return: NotNullIfNotNull("s")]
         public static string? EscapeControlChars(string? s)
         {
-            if (s is not null)
-            {
-                StringBuilder sb = new StringBuilder();
+            int index = 0;
+            return EscapeControlChars(s, ref index);
+        }
 
-                foreach (char c in s)
-                {
-                    switch (c)
-                    {
-                        //case '\'':
-                        //    sb.Append("\\\'");
-                        //    break;
-                        //case '\"':
-                        //    sb.Append("\\\"");
-                        //    break;
-                        case '\\':
-                            sb.Append("\\\\");
-                            break;
-                        case '\0':
-                            sb.Append("\\0");
-                            break;
-                        case '\a':
-                            sb.Append("\\a");
-                            break;
-                        case '\b':
-                            sb.Append("\\b");
-                            break;
-                        case '\f':
-                            sb.Append("\\f");
-                            break;
-                        case '\n':
-                            sb.Append("\\n");
-                            break;
-                        case '\r':
-                            sb.Append("\\r");
-                            break;
-                        case '\t':
-                            sb.Append("\\t");
-                            break;
-                        case '\v':
-                            sb.Append("\\v");
-                            break;
+        /// <summary>
+        /// Converts any control characters in a string
+        /// to their escaped representation.
+        /// </summary>
+        /// <param name="s">The string to be converted</param>
+        /// <param name="index">The index in the array of a specific spot, which needs to be updated when expanding.</param>
+        /// <returns>The converted string</returns>
+        [return: NotNullIfNotNull("s")]
+        public static string? EscapeControlChars(string? s, ref int index)
+        {
+            if (s is null)
+                return null;
 
-                        case '\x0085':
-                        case '\x2028':
-                        case '\x2029':
-                            sb.Append($"\\x{(int)c:X4}");
-                            break;
+            int originalIndex = index;
+            StringBuilder sb = new(s.Length + 32);
```

Comment:
Why 32 ?

Reply:
Purely to prevent an re-allocation/copy in most cases when there are a few characters to escape.
It could be any number. I can make it the magic '42' (_“The Hitchhiker's Guide to the Galaxy”_) or 10.